src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -130,6 +140,103 @@ private bool TryAuthenticateFromCache(
 		return true;
 	}
 
+	private async Task<IOktaAuthenticatedClient?> TryAuthenticateWithDSSOAsync(
+		string org,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool experimentalBypassBrowserSecurity
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimentalBypassBrowserSecurity );
+		if( browser is null ) {
+			return null;
+		}
+
+		Console.WriteLine( "Attempting to automatically login using DSSO." );
+		var cancellationTokenSource = new CancellationTokenSource( TimeSpan.FromSeconds( 15 ) );
+		var sessionIdTaskProducer = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		string? sessionId;
+
+		try {
+			var page = await browser.NewPageAsync().WaitAsync( cancellationTokenSource.Token );
+			string baseAddress = org.Contains( '.' )
+				? $"https://{org}/"
+				: $"https://{org}.okta.com/";
+			var baseUrl = new Uri( baseAddress );
+			int attempt = 1;
+
+			page.Load += ( _, _ ) => _ = GetSessionCookieAsync();
+			await page.GoToAsync( baseAddress );
+			sessionId = await sessionIdTaskProducer.Task.WaitAsync( cancellationTokenSource.Token );
+
+			async Task GetSessionCookieAsync() {
+				var url = new Uri( page.Url );
+				if( url.Host == baseUrl.Host ) {
+					string title = await page.GetTitleAsync();
+					// DSSO can sometimes takes more than one attempt.
+					// If the path is '/', it means DSSO is not available and we should stop retrying.
+					if( title.Contains( "sign in", StringComparison.OrdinalIgnoreCase ) ) {
+						if( attempt < 3 && url.AbsolutePath != "/" ) {
+							attempt++;
+							await page.GoToAsync( baseAddress );
+						} else {
+							sessionIdTaskProducer.SetResult( null );
+						}
+						return;
+					}
+				}
+				var cookies = await page.GetCookiesAsync( baseAddress );
+				if( Array.Find( cookies, c => c.Name == "sid" )?.Value is string sid ) {
+					sessionIdTaskProducer.SetResult( sid );
+				}
+			}
+		} catch( TaskCanceledException ) {
+			consoleWriter.WriteWarning( $"""
+				WARNING: Timed out when trying to create Okta session through DSSO.
+				Check if the org '{org}' is correct. If running BMX with elevated privileges,
+				rerun the command with the '--experimental-bypass-browser-security' flag
+				"""
+			);
+			return null;
+		} catch( TargetClosedException ) {
+			consoleWriter.WriteWarning( """
+				WARNING: Failed to create Okta session through DSSO as BMX is likely being run
+				with elevated privileges. Rerun the command with the '--experimental-bypass-browser-security' flag.
+				"""
+			);
+			return null;
+		} catch( Exception ) {
+			consoleWriter.WriteWarning( "WARNING: Unknown error while trying to authenticate with Okta using DSSO." );
+			return null;
+		} finally {
+			cancellationTokenSource.Dispose();
+			browser.Dispose();
```

Comment:
nit: Browser is being initialized here with using statement syntax https://github.com/Brightspace/bmx/blob/40171e4acf81ad2da538ffe6a1e6f396a8e3186f/src/D2L.Bmx/OktaAuthenticator.cs#L149

The whole point of which is to ensure it will get disposed of properly automatically. Doing this shouldn't be necessary